sources/dogtag/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+/*!
+dogtag is a set of tools that detect the hostname of a bottlerocket server/instance and prints it to stdout.
+if the tool is called in an environment it cannot resolve the hostname it will error out.
+
+Currently the following hostname tools are implemented:
+
+* 01-imds - Fetches hostname from the Instance Metadata via IMDS
+* 00-reverse-dns - Uses reverse dns lookup to resolve the hostname
+ */
+use argh::FromArgs;
+use log::debug;
+use snafu::ResultExt;
+use std::{fs, process};
+
+const DOGTAG_BIN_PATH: &str = "/var/bottlerocket/dogtag";
```

Comment:
for the similar functionality in `early-boot-config`, we use `/usr/libexec/early-boot-config/data-providers.d/`, not sure if it would be more coherent for these lookup binaries to live under `/usr/libexec/dogtag.d/`?

Reply:
I think that makes more sense, will adjust this

Reply:
I'm realizing that there is actually a bit of a distinction here -- `early-boot-config` merges all of the results of its data providers into one, much more in keeping with what you'd expect from a `.d` directory, whereas after any of these hostname resolvers return a result, we return it. I'm ok to stick with the current name